common-npm-packages/packaging-common/nuget/NuGetConfigHelper2.ts

```diff
@@ -64,76 +66,105 @@ export class NuGetConfigHelper2 {
         }
     }
 
-    public addSourcesToTempNuGetConfig(packageSources: IPackageSource[]): void
-    {
+    public addSourcesToTempNuGetConfig(packageSources: IPackageSource[]): void {
         tl.debug('Adding sources to nuget.config');
         this.ensureTempConfigCreated();
         this.addSourcesToTempNugetConfigInternal(packageSources);
     }
 
-    public setAuthForSourcesInTempNuGetConfig(): void
-    {
+    /**
+ * Set authentication for source in a temporary nuget config file.
+ *
+ *
+ * @param nugetVersion - Optional.  If nugetVersion is provided, will use it to check if version is equal to 4.9.2.  Nuget 4.9.2 has a bug that requires a special work around.  If nugetVersion is not provided, will assusm that nuget version is not 4.9.2.
+ * @returns void. This method set authentication for source in a temporary nuget config file.
+ *
+ */
+    public setAuthForSourcesInTempNuGetConfig(nugetVersion?: VersionInfoVersion): void {
         tl.debug('Setting auth in the temp nuget.config');
         this.ensureTempConfigCreated();
 
         let sources = this.getSourcesFromTempNuGetConfig();
-        if (sources.length < 1)
-        {
+        if (sources.length < 1) {
             tl.debug('Not setting up auth for temp nuget.config as there are no sources');
             return;
         }
+        let append = false;
 
+        if (nugetVersion) {
+            if (nugetVersion.a == 4 && nugetVersion.b == 9 && nugetVersion.c == 2) {
+                append = true;
+                tl.debug(`NuGet Version ${nugetVersion.toString()} detected.  Appending "feed-" to the key`);
+            }
+        }
+        const dotnetPath = tl.which('dotnet', false);
+        if (dotnetPath) {
+            try {
+                let dotnet = tl.tool(dotnetPath);
+                dotnet.arg('--version');
+                let version = dotnet.execSync().stdout.trim();
+                let versionArr = version.split(".");
+                if ((versionArr[0] == "2") && (versionArr[1] == "1") && (versionArr[2] == "500")) {
+                    append = true;
+                    tl.debug(`Dotnet Version ${version} detected.  Appending "feed-" to the key`);
+                }
+
+            } catch (err) {
+                tl.debug(err.message);
+                append = false;
+            }
+        }
+        this.setAuthForSourcesInTempNuGetConfigHelper(sources, append);
```

Comment:
Instead of factoring this code out and giving it a duplicate name, factor out the rest of the code above and give a more descriptive name like `validateToolsVersions` or something else to denote special handling based on tools versions